react/features/video-quality/subscriber.js

```diff
@@ -193,43 +193,114 @@ function _updateReceiverVideoConstraints({ getState }) {
     const maxFrameHeight = Math.min(maxReceiverVideoQuality, preferredVideoQuality);
     const { remoteScreenShares } = state['features/video-layout'];
     const { visibleRemoteParticipants } = state['features/filmstrip'];
+    const { flags: { sourceNameSignaling } } = state['features/base/config'];
+    const tracks = state['features/base/tracks'];
 
     const receiverConstraints = {
         constraints: {},
         defaultConstraints: { 'maxHeight': VIDEO_QUALITY_LEVELS.NONE },
         lastN,
-        onStageEndpoints: [],
-        selectedEndpoints: []
+        ...sourceNameSignaling ? { prioritizedSources: [] } : { onStageEndpoints: [] },
+        ...sourceNameSignaling ? { selectedSources: [] } : { selectedEndpoints: [] }
     };
 
-    // Tile view.
-    if (shouldDisplayTileView(state)) {
-        if (!visibleRemoteParticipants?.size) {
-            return;
+    if (sourceNameSignaling) {
+        const participantIdToTrackSourceName = tracks.reduce((acc, { jitsiTrack, participantId }) => {
+            const local = jitsiTrack.isLocal();
+            const isVideo = jitsiTrack && jitsiTrack.type === 'video';
```

Comment:
Is there a better way to determine if a track is video? I also assumed we only want to pull video tracks.